src/main/java/org/takes/rq/RqOnce.java

```diff
@@ -54,22 +52,17 @@ public RqOnce(final Request req) {
      */
     private static Request wrap(final Request req) {
         final AtomicBoolean seen = new AtomicBoolean(false);
-        return new Request() {
-            @Override
-            public Iterable<String> head() throws IOException {
-                return req.head();
-            }
-
-            @Override
-            public InputStream body() throws IOException {
+        return new RequestOf(
+            req::head,
+            () -> {
```

Comment:
@fanifieiev WDYT change this piece of code for the `Sticky` decorator? Thus, multiple `body()` calls return the same content.

Reply:
@fabriciofx I will create a `todo `for that. I do not want to mix that with the current task.